### PR TITLE
Add blackmagic davinci resolve

### PR DESCRIPTION
**Pull request by donbobka**
Add cask for Blackmagic Davinci Resolve - one of the most powerful video production tool.
There are two versions of Davinci Resolve:
- Normal - works only with special usb dongle :moneybag::moneybag::moneybag:.
- Lite - absolutely free, but without some features

What version should place here? I add casks for both.

There is a problem with uninstaller: To uninstall davinci resolve you should execute special app ["Uninstall Resolve.app"](http://content.screencast.com/users/donbobka/folders/Snagit/media/4a2122f3-baea-49ea-a3a7-104bdac0f89c/2015-01-09_03-29-45.png). I implemented uninstall through `uninstall :script`, but when script executed, uninstaller asks user input. Is there something like `installer :manual` for uninstaller?

P.S.: Check Utils.blackmagick_url, please. Is there a more beautiful way?
P.P.S.: Is there way to prevent installation of cask if another cask already installed? For example: if davinci-resolve installed, then you can't install davinci-resolve-lite and vice-versa

### CODE REVIEW

File: Casks/davinci-resolve-lite.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  url Utils.blackmagick_url('https://www.blackmagicdesign.com/api/register/us/download/15061',
+                            'product' => { 'name' => 'DaVinci Resolve Lite' })
+
+  homepage 'https://www.blackmagicdesign.com/products/davinciresolve'
+  license :gratis
+
+  container :nested => "DaVinci_Resolve_Lite_#{version}_Mac.dmg"
+  pkg "Install Resolve #{version} Lite.pkg"
+
+  uninstall :script => { :executable => '/usr/bin/open', :args => ['/Applications/DaVinci Resolve/Uninstall Resolve.app'] }
```
Comment (vitorgalvao) [inline Casks/davinci-resolve-lite.rb]: Your misunderstanding how this works. It doesn’t mean “write a script to uninstall”, it means “give a path to an uninstall script the vendor provided”.

Comment (donbobka) [inline Casks/davinci-resolve-lite.rb]: Like that?
```ruby
uninstall :script => '/Applications/DaVinci Resolve/Uninstall Resolve.app/Contents/MacOS/Uninstall Resolve'
```

**Comment by vitorgalvao** [comment]
We should include neither. It’s one thing when the vendor puts up a registration wall but the url is publicly accessible, but it’s quite another to load an entire external library and fake an identity to bypass said wall. To me, that is highly unreasonable. If the developer goes to such lengths and puts such technical walls in front of its downloads, we shouldn’t go around them.

Thank you nonetheless for the work you put into this, @donbobka. I’m also leave it open for now, in case other maintainers want to chime in, by I say we close this.

File: Casks/davinci-resolve-lite.rb (hunk 1 above)
Comment (vitorgalvao) [inline Casks/davinci-resolve-lite.rb]: It’s only applicable in certain situations (see [vagrant.rb](https://github.com/caskroom/homebrew-cask/blob/master/Casks/vagrant.rb)), when the vendor specifically provided a script.

**Comment by donbobka** [comment]
In really, fake identity required only for lite version. Let's i change davinci-resolve.rb

**Comment by vitorgalvao** [comment]
Yes, but the non-lite version has a huge link with a token and an expiration, so it’s no good either.

**Comment by donbobka** [comment]
The same way like in [x-mirage.rb](https://github.com/caskroom/homebrew-cask/blob/master/Casks/x-mirage.rb)

**Comment by vitorgalvao** [comment]
What do you mean? `x-mirage` has a tiny url that can be easily `curl`ed and doesn’t expire.

**Comment by donbobka** [comment]
The same way like in [displaylink.rb](https://github.com/caskroom/homebrew-cask/blob/6a66da2e01e5d9b7b71921045aaeb810e22ff206/Casks/displaylink.rb). They use a amazon s3 to store files

**Comment by vitorgalvao** [comment]
Fair point (although the where they actually host the files is irrelevant). You’re still loading an entire library for a single action, though, which wasn’t addressed. Can you do it in the same manner as the `displaylink.rb` cask?

**Comment by donbobka** [comment]
Fast answer: no, because displaylink returns response with http redirect to amazon s3(with token) and blackmagick only returns link(with token) in response body, which we should process and make second request. But i'll try to find some workaround.

**Comment by vitorgalvao** [comment]
Closing for lack of further interest/implementation.

**Pull request closed without merging**